### PR TITLE
[devel/pkg] Update the devel packaging 

### LINKED ISSUE #986: C++ API and architecture refactor
This issue is to track the re-factor of the exposed headers and the C++ API.
This will update the NNTrainer architecture to the class diagram finalized with @jijoongmoon.

- [x] Update internal implementation to use only the methods exposed by those headers and not an internal implementation
  - [x] Remove all friends of layers
  - [x] Manage input/output properties better for layer - track at #1454
  - [x] add unittests for ensureName related  to #1009
  - [x] refactor graph to move graph node related members from layers
    - [x] #1029 
    - [x] consider removing adj lists once compile is done to save some space. - #1049
    - [x] move layers connection related properties out to graph - #1142 #1182
  - [x] tensor constructor to support initialization - https://github.com/nnstreamer/nntrainer/pull/1031#discussion_r597551354 - pointed out by @myungjoo - #1452
  - [x] refactor model loader to allow custom properties #1032 
  - [x] prepare custom loss #1256
  - [x] many more
- [x] Update headers and their interfaces
  - [x] Make c++ api equivalent to c API - done with #1050
  - [x] C++ API must be superset of https://github.com/nnstreamer/nntrainer/pull/1010#discussion_r594866783 or the nntrainer filter must be updated to accommodate with the API. - #1461
  - [x] Make c api directly use c++ API - #1050
  - [x] divide `*_internal` to `*_devel` which will be just an interface and `*_impl` which will be the default implementation. For example, Layer <- Layer_devel <- Layer_impl. - #1426
- [ ] Update packaging and applications correspondingly #1450

### PR DESCRIPTION
**Pull request by kparichay**
Update the devel packaging to remove unnecessary headers.. Below is the list of packaging update:
- weight/var_grad headers are removed
- app_context and other graph, network related headers are removed. They
can be added once network/neural network headers are refactored for
custom models.
- manager and dynamic_training_optimization header is removed
- databuffer headers are commented. They will be updated once dataset
refactoring is done.

Resolves #986

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Comment by kparichay** [comment]
Please note that this patch will be breaking any independent applications code which uses internal headers directly.

File: debian/nntrainer-dev.install [hunk 1]
```diff
@@ -17,6 +17,10 @@
 /usr/include/nntrainer/layer_devel.h
 /usr/include/nntrainer/neuralnet.h
 /usr/include/nntrainer/tensor.h
+<<<<<<< HEAD
+=======
+/usr/include/nntrainer/tensor_dim.h
+>>>>>>> [weight/var_grad] Remove exposure of weight/var_grad
```
Comment (zhoonit) [inline debian/nntrainer-dev.install]: Could you check this out?

Comment (zhoonit) [reply debian/nntrainer-dev.install]: I can see updated, maybe deleting this commit?

**Review by zhoonit — APPROVED** [review]
LGTM with minor issue :)

File: debian/nntrainer-dev.install (hunk 1 above)
Comment (kparichay) [reply debian/nntrainer-dev.install]: done.

File: packaging/nntrainer.spec [hunk 2]
```diff
@@ -431,33 +431,26 @@ cp -r result %{buildroot}%{_datadir}/nntrainer/unittest/
...
 %{_includedir}/nntrainer/databuffer_factory.h
+# layer headers
 %{_includedir}/nntrainer/layer_context.h
 %{_includedir}/nntrainer/layer_devel.h
-%{_includedir}/nntrainer/neuralnet.h
-%{_includedir}/nntrainer/tensor.h
-%{_includedir}/nntrainer/tensor_wrap_specs.h
+%{_includedir}/nntrainer/layer_impl.h
+# optimizer headers
+%{_includedir}/nntrainer/optimizer_context.h
```
Comment (jijoongmoon) [inline packaging/nntrainer.spec:451]: Great!. Looks like more organized and simpler!

**Review by jijoongmoon — APPROVED** [review]
LGTM!

**Pull request merged**